### DISCUSSION TITLE
Lazydev alternative?

### DISCUSSION THREAD

**Discussion by benlubas** [lumen-oss/rocks.nvim#488] [Q&A] [answered]
When I use lazy.nvim I use lazydev to get completions for `vim.api` etc.

I'm unable to use lazydev with rocks.nvim, and I'm wondering what you use instead? @vhyrro seems to have lazydev installed anyway, but I'm really not sure how the plugin is doing anything for him.

There's a possibility that I'm configuring lazydev wrong I guess. But this section from lazydev's "requirements" makes me think that the plugin just won't work with rocks.nvim.

> - [lazy.nvim](https://github.com/folke/lazy.nvim) plugin manager
  	- OR a plugin manager that uses Neovim's native package system

**Comment by mrcjkb** [comment, marked as answer]
rocks.nvim creates symlinks in the packpath, so it works with things that rely on neovim's native packaging system. 

Are you using nvim-lspconfig?
If not, there's also this:

> If you don't use nvim-lspconfig, then you can use require('lazydev').find_workspace(buf?) to check if the buffer is part of an existing workspace or its libraries.

I can try it out later. 

Personally, I use [this](https://github.com/mrcjkb/nix-gen-luarc-json). But it uses Nix and I don't care about lazily updating the workspace libraries.

**Reply by benlubas** [reply]
Okay I think the issue is the setup function not getting called either at all or at the right time.

I have tried `config = {}` and `config = { integrations = { cmp = true } }` (a default option) both with and without `ft = "lua"` from rocks-lazy, and none of them give me completion/hover. But when I run `:= require("lazydev").setup()` after opening a lua file, it starts to work.

Does `config =` not work with rocks-lazy yet?

edit:

> both with and without `ft = "lua"`

I think I might have been impatient when testing in the past. when I don't use `ft = lua` thinks work just fine.

**Reply by benlubas** [reply]
And an answer to myself. `load_opt_plugins = true` needs to be set I see now.

I find the wording in that README a bit confusing, and this option name seems to suggest there are unintended side effects here, but I guess I will deal with them later

**Reply by mrcjkb** [reply]
Oops. There was a bug in rocks-lazy.nvim that broke the rocks-config integration :facepalm: 
I've just pushed a fix.

You shouldn't need to set `load_opt_plugins = true`. That's only if you need to mark plugins as `opt`, but for some reason want to eagerly load the config.
